app/services/sync_ndbn_members.rb

```diff
@@ -1,69 +1,45 @@
 class SyncNDBNMembers
-  include HTTParty
+  include ServiceObjectErrorsMixin
 
-  NDBN_MEMBERS_PAGE = "https://ndbn.memberclicks.net/member-ids".freeze
+  def initialize(member_file)
+    @member_file = member_file
+  end
+  attr_reader :member_file
 
-  class << self
-    def sync
-      Rails.logger.info("Syncing NDBNMember started.")
+  def call
+    member_entries = parse_csv
 
-      members_page = fetch_members_page
-      member_entries = extract_member_entries(members_page)
+    return if errors.any?
 
-      member_entries.each do |member|
-        ndbn_member = NDBNMember.find_or_initialize_by(ndbn_member_id: member[:ndbn_member_id])
-        ndbn_member.account_name = member[:account_name]
+    member_entries.each do |member_id, account_name|
+      ndbn_member = NDBNMember.find_or_initialize_by(ndbn_member_id: member_id.to_i)
+      ndbn_member.account_name = account_name
 
-        # Skip if nothing has changed!
-        if ndbn_member.persisted? && !ndbn_member.changed?
-          next
-        elsif ndbn_member.persisted? && ndbn_member.changed?
-          Rails.logger.info("#{ndbn_member.id} changed their account name from #{ndbn_member.account_name_was} to #{ndbn_member.account_name}.")
-        elsif ndbn_member.new_record?
-          Rails.logger.info("New record found! #{ndbn_member.id} - #{ndbn_member.account_name} has been added!")
-        end
+      # Skip if nothing has changed!
+      if ndbn_member.persisted? && !ndbn_member.changed?
+        next
+      end
 
-        Rails.logger.info("Updating....")
-        if ndbn_member.save
-          Rails.logger.info("Done!")
-        else
-          error_msg = "Failed to update #{ndbn_member_id} due to #{e.errors.full_messages}"
-          Rails.logger.info(error_msg)
-          Bugsnag.notify(error_msg)
+      unless ndbn_member.save
+        ndbn_member.errors.full_messages.each do |msg|
+          error = "Issue with #{member_id}: #{account_name} -> #{msg}"
+          errors.add(:base, error)
         end
       end
-
-      Rails.logger.info("Syncing NDBNMember finished.")
     end
+  end
 
-    private
-
-    def fetch_members_page
-      res = get(NDBN_MEMBERS_PAGE)
-
-      if res.code != 200
-        raise "SyncNDBNMembers.sync failed due to getting a status code of #{res.code}"
-      end
+  private
 
-      res.body
-    end
+  def parse_csv
+    return @_parsed if defined?(@_parsed)
```

Comment:
Do we have a reason to think we'd be calling this method multiple times in a row?